[JSC] Update RegExp UCD to version 15.0.0

msaboff opened the pull request:
#### 66329834d4f68f6ff4ed91a468886594ae9ba301
<pre>
[JSC] Update RegExp UCD to version 15.0.0
<a href="https://bugs.webkit.org/show_bug.cgi?id=252633">https://bugs.webkit.org/show_bug.cgi?id=252633</a>
rdar://105702434

Reviewed by Yusuke Suzuki.

Updated the UCD files used to process Unicode properties in JavaScript regular expressions.

* JSTests/test262/expectations.yaml: Updated for new passing RegExp property escape tests.
* Source/JavaScriptCore/ucd/CaseFolding.txt:
* Source/JavaScriptCore/ucd/DerivedBinaryProperties.txt:
* Source/JavaScriptCore/ucd/DerivedCoreProperties.txt:
* Source/JavaScriptCore/ucd/DerivedNormalizationProps.txt:
* Source/JavaScriptCore/ucd/PropList.txt:
* Source/JavaScriptCore/ucd/PropertyAliases.txt:
* Source/JavaScriptCore/ucd/PropertyValueAliases.txt:
* Source/JavaScriptCore/ucd/ScriptExtensions.txt:
* Source/JavaScriptCore/ucd/Scripts.txt:
* Source/JavaScriptCore/ucd/UnicodeData.txt:
* Source/JavaScriptCore/ucd/emoji-data.txt:
* Source/JavaScriptCore/yarr/generateYarrUnicodePropertyTables.py:

Canonical link: <a href="https://commits.webkit.org/260607@main">https://commits.webkit.org/260607@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/11ee450f10cd1963e57ecb1fcfc208ba9e07735d

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/108798 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/17897 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/41629 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/87/builds/123 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/118097 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/19351 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/9174 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/101026 "Built successfully") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/114566 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/14516 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/97731 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/42488 "Passed tests") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/96489 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/29369 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/84338 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/1/builds/97903 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/10679 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/30720 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/98715 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/83/builds/8805 "Built successfully and passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/11440 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/7653 "Passed tests") | [  ~~🧪 jsc-armv7-tests~~](https://ews-build.webkit.org/#/builders/46/builds/30982 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/16827 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/50311 "Passed tests") | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/106331 "Built successfully") | 
| [❌ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/7328 "Failed to push commit to Webkit repository") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/13027 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/45/builds/26356 "Passed tests") | 
| | | | | 
<!--EWS-Status-Bubble-End-->

Constellation reviewed (approved):
Can you check whether it pass some more test262 tests?

msaboff committed:
[JSC] Update RegExp UCD to version 15.0.0
https://bugs.webkit.org/show_bug.cgi?id=252633
rdar://105702434

Reviewed by Yusuke Suzuki.

Updated the UCD files used to process Unicode properties in JavaScript regular expressions.

* JSTests/test262/expectations.yaml: Updated for new passing RegExp property escape tests.
* Source/JavaScriptCore/ucd/CaseFolding.txt:
* Source/JavaScriptCore/ucd/DerivedBinaryProperties.txt:
* Source/JavaScriptCore/ucd/DerivedCoreProperties.txt:
* Source/JavaScriptCore/ucd/DerivedNormalizationProps.txt:
* Source/JavaScriptCore/ucd/PropList.txt:
* Source/JavaScriptCore/ucd/PropertyAliases.txt:
* Source/JavaScriptCore/ucd/PropertyValueAliases.txt:
* Source/JavaScriptCore/ucd/ScriptExtensions.txt:
* Source/JavaScriptCore/ucd/Scripts.txt:
* Source/JavaScriptCore/ucd/UnicodeData.txt:
* Source/JavaScriptCore/ucd/emoji-data.txt:
* Source/JavaScriptCore/yarr/generateYarrUnicodePropertyTables.py:

Canonical link: https://commits.webkit.org/260607@main
